clients/python/src/model_registry/core.py

```diff
@@ -2,45 +2,109 @@
 
 from __future__ import annotations
 
-from ml_metadata.proto import MetadataStoreClientConfig
+import os
+from dataclasses import dataclass
+from pathlib import Path
+from warnings import warn
+
+import grpc
 
 from .exceptions import StoreException
 from .store import MLMDStore, ProtoType
 from .types import ListOptions, ModelArtifact, ModelVersion, RegisteredModel
 from .types.base import ProtoBase
 from .types.options import MLMDListOptions
+from .utils import header_adder_interceptor
 
 
+@dataclass
 class ModelRegistryAPIClient:
     """Model registry API."""
 
-    def __init__(
-        self,
+    store: MLMDStore
+
+    @classmethod
+    def secure_connection(
+        cls,
+        server_address: str,
+        port: int = 443,
+        user_token: bytes | None = None,
+        custom_ca: bytes | None = None,
+    ) -> ModelRegistryAPIClient:
+        """Constructor.
+
+        Args:
+            server_address: Server address.
+            port: Server port. Defaults to 443.
+            user_token: The PEM-encoded user token as a byte string. Defaults to envvar KF_PIPELINES_SA_TOKEN_PATH.
+            custom_ca: The PEM-encoded root certificates as a byte string. Defaults to envvar CERT.
+        """
+        if not custom_ca:
+            ca_cert = os.environ.get("CERT")
+            if not ca_cert:
+                msg = "CA certificate must be provided"
+                raise StoreException(msg)
+            root_certs = Path(ca_cert).read_bytes()
+        else:
+            root_certs = custom_ca
```

Comment:
Since `secureconnection` or `insecureconnection` is based on this block, I would pull this up before the connection is determined. There is no need to check port is 443 or not IMO. so, such as

```
if root_certs:
   conn = buildSecureConnection
else:
  conn = insuecureConnection
```